### PR TITLE
Add Krona output for Kraken classification

### PR DESCRIPTION
**Pull request by MillironX**
Adds the

- KRAKENTOOLS_KRAKEN2KRONA
- KRONA_KTIMPORTTEXT

modules from nf-core/modules, and

- KRONA_CLEANUP

a local module to reformat kraken2krona's output to make it pretty.

Then adds these tools to the PROFILING subworkflow and creates one Krona chart per Kraken database.

Right now, only Kraken gets Krona output, but it should be easy to add others soon.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/taxprofiler/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/taxprofiler _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by jfy133** [comment]
Thanks very much for this @MillironX - I 'm unfortnuately getting snowed under this week so can't look properly until next week.

But as I expect we will have multiple Krona conversion modules, I think it would make sense maybe to have a separate post_processing krona workflow? Does this makes sense from your point of view (I've only used Krona a long time ago and for a single tool...)

**Comment by MillironX** [comment]
Yes, that makes sense. It will take some time to figure out how to route taxonomy reports from each of the profilers through converters and to Krona. Adding more classifier converters seemed out-of-scope for a single PR, but I can convert this to draft and investigate what the overall interface would need to look like, if you would prefer.

File: subworkflows/local/visualization.nf [hunk 1]
```diff
@@ -0,0 +1,56 @@
+//
+// Create Krona visualizations
+//
+
+include { KRAKENTOOLS_KREPORT2KRONA } from '../../modules/nf-core/modules/krakentools/kreport2krona/main'
+include { KRONA_CLEANUP } from '../../modules/local/krona_cleanup'
+include { KRONA_KTIMPORTTEXT } from '../../modules/nf-core/modules/krona/ktimporttext/main'
+
+
```
Comment (jfy133) [inline subworkflows/local/visualization.nf:9]:
```suggestion
```

File: subworkflows/local/visualization.nf [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        .branch {
+            kraken2: it[0]['tool'] == 'kraken2'
+            unknown: true
+        }
+
+    /*
+        Convert Kraken2 formatted reports into Krona text files
+    */
+    ch_kraken_reports = ch_input_profiles.kraken2
+    KRAKENTOOLS_KREPORT2KRONA ( ch_kraken_reports )
```
Comment (jfy133) [inline subworkflows/local/visualization.nf:32]:
```suggestion
    KRAKENTOOLS_KREPORT2KRONA ( ch_input_profiles.kraken2 )
```

File: subworkflows/local/visualization.nf [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    ch_krona_text = ch_krona_text.mix( KRAKENTOOLS_KREPORT2KRONA.out.txt )
+    ch_versions = ch_versions.mix( KRAKENTOOLS_KREPORT2KRONA.out.versions.first() )
+
+    /*
+        Remove taxonomy level annotations from the Krona text files
+    */
+    KRONA_CLEANUP( ch_krona_text )
```
Comment (jfy133) [inline subworkflows/local/visualization.nf:39]: Does this apply to all modules or just kraken?

File: workflows/taxprofiler.nf [hunk 4]
```diff
@@ -209,6 +210,12 @@ workflow TAXPROFILER {
     PROFILING ( ch_reads_runmerged, DB_CHECK.out.dbs )
     ch_versions = ch_versions.mix( PROFILING.out.versions )
 
+    /*
+        SUBWORKFLOW: VISUALIZATION
+    */
+    VISUALIZATION ( PROFILING.out.profiles )
```
Comment (jfy133) [inline workflows/taxprofiler.nf:216]: I would like to make this optional, so maybe wrap it with an if statement and evaluating a new  `params.run_visualisation_krona` or something.

File: subworkflows/local/profiling.nf [hunk 5]
```diff
@@ -2,15 +2,15 @@
...
-include { KAIJU_KAIJU2TABLE           } from '../../modules/nf-core/modules/kaiju/kaiju2table/main'
-include { DIAMOND_BLASTX              } from '../../modules/nf-core/modules/diamond/blastx/main'
+include { MALT_RUN                                 } from '../../modules/nf-core/modules/malt/run/main'
+include { MEGAN_RMA2INFO                           } from '../../modules/nf-core/modules/megan/rma2info/main'
+include { KRAKEN2_KRAKEN2                          } from '../../modules/nf-core/modules/kraken2/kraken2/main'
+include { CENTRIFUGE_CENTRIFUGE                    } from '../../modules/nf-core/modules/centrifuge/centrifuge/main'
+include { CENTRIFUGE_KREPORT                       } from '../../modules/nf-core/modules/centrifuge/kreport/main'
+include { METAPHLAN3                               } from '../../modules/nf-core/modules/metaphlan3/main'
+include { KAIJU_KAIJU                              } from '../../modules/nf-core/modules/kaiju/kaiju/main'
+include { KAIJU_KAIJU2TABLE                        } from '../../modules/nf-core/modules/kaiju/kaiju2table/main'
```
[the remark this answers is not in the document]
Comment (MillironX) [reply subworkflows/local/profiling.nf]: All changes to `profiling.nf` have been rolled back.

File: subworkflows/local/visualization.nf (hunk 2 above)
Comment (MillironX) [reply subworkflows/local/visualization.nf]: This is done this way so I can add centrifuge next

File: subworkflows/local/visualization.nf (hunk 3 above)
Comment (MillironX) [reply subworkflows/local/visualization.nf]: All modules

File: workflows/taxprofiler.nf (hunk 4 above)
Comment (MillironX) [reply workflows/taxprofiler.nf]: See 89d4d1a

**Review by jfy133 — APPROVED** [review]
Perfect, thank you very much @MillironX !

**Pull request merged**